pkg/controller/cluster/main-controller.go

```diff
@@ -500,6 +517,18 @@ func (c *Controller) Start(threadiness int, stopCh <-chan struct{}) error {
 	leaseLockName := "minio-operator-lock"
 	leaseLockNamespace := miniov2.GetNSFromFile()
 
+	// stsNotificationChannel is a channel for the STS Server API
+	stsNotificationChannel := make(chan error)
```

Comment:
no one writes to this channel when `IsSTSEnabled()` is not true why can't we have a single channel here with indications on what to do,

This would simplify the channel handling to something like 

```
for oerr := range errorCh {
     switch oerr.Type {
     case stsNotification:
       if oerr.Err != nil && oerr.Err != http.ErrServerClosed {
	   klog.Errorf("API Server stopped: %v, restarting..", err)
	   go startSTSAPIServer(ctx, c, errorCh)
       }
     case leaderElectionNotification:
           return nil
     }
}

Reply:
Thank you, will need fresh eyes to get it, and review it tomorrow, but appreciate the feedback

Reply:
created the typed channel @harshavardhana